### PR TITLE
Added missing throw in change password process

### PR DESCRIPTION
**Pull request by core23**
I am targetting this branch, because this is a bugfix.

## Changelog

```markdown
### Fixed
 - Added missing throw in change password process
```

## Subject

The exception was created but not thrown to the request handler for post processing.

### CODE REVIEW

File: Controller/ChangePasswordFOSUser1Controller.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public function changePasswordAction()
     {
         $user = $this->getUser();
         if (!is_object($user) || !$user instanceof UserInterface) {
-            $this->createAccessDeniedException('This user does not have access to this section.');
+            throw $this->createAccessDeniedException('This user does not have access to this section.');
```
Comment (jordisala1991) [inline Controller/ChangePasswordFOSUser1Controller.php]: This change is correct, but there is a problem... this method we are calling is not present on SF < 2.5:

http://symfony.com/blog/new-in-symfony-2-5-create-the-correct-denied-http-exception

IMHO we could simply replace this call by:

`throw new AccessDeniedException('This user does not have access to this section.')`

but there could be other ways like implement this method as a fallback for SF 2.3:

https://github.com/symfony/symfony/blob/2.5/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L166

Or do an `if` to check if the method exists.

Comment (greg0ire) [inline Controller/ChangePasswordFOSUser1Controller.php]: I'd go with the simplest solution, which is not deprecated.

Comment (core23) [inline Controller/ChangePasswordFOSUser1Controller.php]: We're calling this method any three different classes at the moment:
https://github.com/sonata-project/SonataUserBundle/search?utf8=%E2%9C%93&q=createAccessDeniedException

Comment (jordisala1991) [inline Controller/ChangePasswordFOSUser1Controller.php]: Then I would just use `throw new AccessDeniedException('This user does not have access to this section.')` on all of them

Comment (core23) [inline Controller/ChangePasswordFOSUser1Controller.php]: Fixed that

**Comment by jordisala1991** [comment]
Thanks @core23

**Pull request merged**